tools/doc/common.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const yaml = require('js-yaml');
+
+function isYAMLBlock(text){
+  return !!text.match(/^<!-- YAML/);
+}
+
+exports.isYAMLBlock = isYAMLBlock;
+
+function extractAndParseYAML(text){
+  text = text.trim();
+
+  text = text.replace(/^<!-- YAML/, '')
+             .replace(/-->$/, '');
+
+  try {
+    return yaml.safeLoad(text);
+  }catch(e){
+    throw e;
```

Comment:
Why the catch and re-throw? Maybe just leave a comment that safeLoad() throws instead?